clientlibs/js/src/ApiService/ApiService.ts

```diff
@@ -0,0 +1,318 @@
+    }
+  }
+
+  public async init(
+    group?: Record<string, Array<string>>,
+    workingGroup?: Record<string, string>
+  ): Promise<UpGradeClientInterfaces.IExperimentUser> {
+    let requestBody: UpGradeClientInterfaces.IExperimentUser = {
+      id: this.userId,
+    };
+
+    if (group) {
+      requestBody = {
+        ...requestBody,
+        group,
+      };
+    }
+
+    if (workingGroup) {
+      requestBody = {
+        ...requestBody,
+        workingGroup,
+      };
+    }
+
+    return await this.sendRequest<UpGradeClientInterfaces.IExperimentUser>({
+      path: this.api.init,
+      method: UpGradeClientEnums.REQUEST_METHOD.POST,
+      body: requestBody,
+    });
+  }
+
+  public async setGroupMembership(
+    group: UpGradeClientInterfaces.IExperimentUserGroup
+  ): Promise<UpGradeClientInterfaces.IExperimentUser> {
+    const requestBody: UpGradeClientRequests.ISetGroupMembershipRequestBody = {
+      id: this.userId,
+      group,
+    };
+
+    return await this.sendRequest<UpGradeClientInterfaces.IExperimentUser>({
+      path: this.api.setGroupMemberShip,
+      method: UpGradeClientEnums.REQUEST_METHOD.PATCH,
+      body: requestBody,
+    });
+  }
+
+  public async setWorkingGroup(
+    workingGroup: UpGradeClientInterfaces.IExperimentUserWorkingGroup
+  ): Promise<UpGradeClientInterfaces.IExperimentUser> {
+    const requestBody: UpGradeClientRequests.ISetWorkingGroupRequestBody = {
+      id: this.userId,
+      workingGroup,
+    };
+
+    return await this.sendRequest<UpGradeClientInterfaces.IExperimentUser>({
+      path: this.api.setWorkingGroup,
+      method: UpGradeClientEnums.REQUEST_METHOD.PATCH,
+      body: requestBody,
+    });
+  }
+
+  public async setAltUserIds(
+    altUserIds: UpGradeClientInterfaces.IExperimentUserAliases
+  ): Promise<UpGradeClientInterfaces.IExperimentUserAliases> {
+    const requestBody: UpGradeClientRequests.ISetAltIdsRequestBody = {
+      userId: this.userId,
+      aliases: altUserIds,
+    };
+
+    return await this.sendRequest<UpGradeClientInterfaces.IExperimentUser>({
+      path: this.api.altUserIds,
+      method: UpGradeClientEnums.REQUEST_METHOD.PATCH,
+      body: requestBody,
+    });
+  }
+
+  public async getAllExperimentConditions(): Promise<IExperimentAssignmentv5[]> {
+    const requestBody: UpGradeClientRequests.IGetAllExperimentConditionsRequestBody = {
+      userId: this.userId,
+      context: this.context,
+    };
+
+    return await this.sendRequest<IExperimentAssignmentv5[]>({
+      path: this.api.getAllExperimentConditions,
+      method: UpGradeClientEnums.REQUEST_METHOD.POST,
+      body: requestBody,
+    });
+  }
+
+  public async markDecisionPoint({
+    site,
+    target,
+    condition,
+    status,
+    uniquifier,
+    clientError,
+  }: IMarkDecisionPointParams): Promise<UpGradeClientInterfaces.IMarkDecisionPoint> {
+    const assignment = this.dataService.findExperimentAssignmentBySiteAndTarget(site, target);
+
+    if (!assignment) {
```

Comment:
We can mark a decision point without an experiment also. I don't think we should throw error here. Correct me if I am missing something here.

Reply:
hm yeah let me make sure this works if I remove it, this is preexisting from the within-subjects PR

Reply:
I addressed other comments but still need to look into this one tomorrow, definitely seems like you're right but need to look back at the within-subjects work and see if it's something I messed up or if there was a pre-existing bug (or both).

Reply:
so I think there might be a pre-existing issue here, I don't know how we're supposed to mark a decision point without an assignment (aka no experiment running) in the v5 `/mark` API. If that is indeed that case, can that be handled in a separate bugfix PR?

I think there may need to be additional API and client-lib changes if so, currently this code is not right but the original code change for within-subjects before I rearranged things can be seen here: (https://github.com/CarnegieLearningWeb/UpGrade/commit/43d20a7cc16acb47555c2d5e483c9fe23a7918cf#diff-a4a1d40bc005562c8bb6ebb760ded881ff488ebd3bd499ce37213c9457728d2eL30) ... it would throw an exception if assignment is undefined, but that should not stop us from marking.

Reply:
@danoswaltCL can you create a separate issue for this and we can track it there?

Reply:
@VivekFitkariwala #999 and #1000 created.